pkg/expression/builtin_time.go

```diff
@@ -6232,6 +6232,11 @@ func addUnitToTime(unit string, t time.Time, v float64) (time.Time, bool, error)
 			return tb, true, nil
 		}
 		tb = t.AddDate(0, int(v), 0)
+
+		// For corner case: timestampadd(month,1,date '2024-01-31') = "2024-02-29", timestampadd(month,1,date '2024-01-30') = "2024-02-29"
+		for tb.Month() != t.Month()+1 {
```

Comment:
What if add more than 1 month?

Reply:
> What happen if adding more than 1 month?

This is mysql: timestampadd(month,1,date '2024-01-31') = "2024-02-29"

This is tidb: timestampadd(month,1,date '2024-01-31') = "2024-03-02"

The month field of the result of `timestampadd(month,1,date '2024-01-31')` is always 1 more than the month field of input. Adding more than 1 month is unexpected.